File: 1114-print-in-order/1114-print-in-order.java
class Foo {
    private volatile int count;
    public Foo() {
        count = 0;
    }

    public void first(Runnable printFirst) throws InterruptedException {
        
        // printFirst.run() outputs "first". Do not change or remove this line.
        printFirst.run();
        count++;
    }

    public void second(Runnable printSecond) throws InterruptedException {
        
        // printSecond.run() outputs "second". Do not change or remove this line.
        while(count!=1) continue;
        printSecond.run();
        count++;
    }

    public void third(Runnable printThird) throws InterruptedException {
        
        // printThird.run() outputs "third". Do not change or remove this line.
        while(count!=2) continue;
        printThird.run();
    }
}

File: 1114-print-in-order/1114-print-in-order.py
class Foo:
    def __init__(self):
        self.count = 0


    def first(self, printFirst: 'Callable[[], None]') -> None:
        
        # printFirst() outputs "first". Do not change or remove this line.
        printFirst()
        self.count += 1


    def second(self, printSecond: 'Callable[[], None]') -> None:
        
        # printSecond() outputs "second". Do not change or remove this line.
        while self.count != 1:
            continue
        printSecond()
        self.count += 1


    def third(self, printThird: 'Callable[[], None]') -> None:
        
        # printThird() outputs "third". Do not change or remove this line.
        while self.count != 2:
            continue
        printThird()